### PR TITLE
Get the scenario_id from the addon content list sent by the server if it's not otherwise available.

### LINKED ISSUE #6285: Multiplayer scenarios with custom terrains don't load for non-host players and observers
### Game and System Information
 - What version of the game are you running? 1.16.0 on flatpak on Ubuntu 20.04, my friends used Steam on Windows

### Describe the bug
When you start a game in a multiplayer scenario with custom terrains, other players (not the host) see an error window "the map could not be loaded: Unknow tile in the map <some custom terrain id>" and fail. The host, on the other hand, works fine. The same error sees everyone trying to observe such a game. (NB: all the folks have the same version of the addon in question)

### To Reproduce

1. Install any addon with a multiplayer scenario with custom terrains
2. In the lobby Create Game > Scenarios, choose a scenario in question
3. Either start a game with several people or start the game alone and let other people try to observe this game

Addons and scenarios you can try (from the 1.16 addon server): Era of Magic and scenario EoMa 4p - Zombies Challenge; Legend of the Invincibles (beta) and scenario LotI - Gladiators

### Screenshots
GladiatorsRIP
What you see for non-host players/observer in LotI - Gladiators

ZCbug

What you see in EoMa Zombies Challenge

### Additional context
Dunno if the bug is applicable to other things (campaigns, mp campaigns, etc), only tested for separate mp scenarios

### PR DESCRIPTION
**Pull request by Pentarctagon**
Fixes #6285

### CODE REVIEW

**Comment by soliton-** [comment]
So do we need similar logic for replays?

File: src/server/wesnothd/server.cpp [hunk 1]
```diff
@@ -1760,9 +1760,9 @@ void server::handle_player_in_game(player_iterator p, simple_wml::document& data
 		g.level().root().apply_diff(*scenario_diff);
 		const simple_wml::node* cfg_change = scenario_diff->child("change_child");
 
-		// it is very likeley that the diff changes a side so this check isn't that important.
+		// it is very likely that the diff changes a side so this check isn't that important.
 		// Note that [side] is not at toplevel but inside [scenario] or [snapshot]
```
Comment (soliton-) [inline src/server/wesnothd/server.cpp:1764]: The comments seem to be obsolete now.

Comment (Pentarctagon) [reply src/server/wesnothd/server.cpp:1764]: Alright, I'll remove it then.

**Comment by soliton-** [comment]
That probably means for replays all addons are loaded which presumably means potential for OOS if when playing and replaying different WML is used.

**Comment by Pentarctagon** [comment]
If that is the case, I think that would be a separate enhancement, beyond the currently opened issue.

**Pull request merged**